client/state/preferences/actions.js

```diff
@@ -0,0 +1,52 @@
+/**
+ * Internal dependencies
+ */
+import wpcom from 'lib/wp';
+import {
+	PREFERENCES_SET,
+	PREFERENCES_RECEIVE,
+	PREFERENCES_REMOVE,
+	PREFERENCES_FETCH
+} from 'state/action-types';
+
+export const USER_SETTING_KEY = 'calypso_preferences';
+
+export function fetchPreferences() {
+	return ( dispatch ) => {
+		dispatch( { type: PREFERENCES_FETCH } );
+		return wpcom.me().settings().get().then( ( data ) => {
+			dispatch( {
+				type: PREFERENCES_RECEIVE,
+				data
+			} );
+		} );
+	};
+}
+
+const save = () => ( dispatch, getState ) => {
```

Comment:
I didn't expect `save` to be used in any use case, that's why it was not exported! :)
This is an utililty function that was used in `set` and `remove`.

- I reoved `remove`
- I moved save logic to `set` and removed `save`

But `getState` stays.
Because endpoint works how I explained in this comment: https://github.com/Automattic/wp-calypso/pull/5401#discussion_r63869233 , to save one change in the endpoint, we have to provide serialized new state of all other settings. We can either do `getState` or force developers to first manually read state, amend and save. Which can go HORRIBLY wrong.
It also does not harm readability in this case, it provides the abstraction that `me/settings` `calypso_settings` is key/value store which in reality, is not.